### PR TITLE
[Core] Support the reduction of Port rate limit in the integrations

### PR DESCRIPTION
**Pull request by shalev007**
# Description

What - I updated the retry mechanism to ensure the total retry time is capped at 5 minutes, with a maximum backoff time of 5 minutes between retries.

Why - This change is needed to handle the reduced port rate limit in integrations more effectively, minimizing the risk of overwhelming the system while still ensuring retries are reasonable.

How - I adjusted the retry logic to distribute retries over the 5-minute window, scaling backoff times to align with the reduced rate limit without exceeding the time limit.

## Type of change

Please leave one option from the following and delete the rest:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] New Integration (non-breaking change which adds a new integration)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [X] Non-breaking change (fix of existing functionality that will not change current behavior)
- [ ] Documentation (added/updated documentation)

<h4> All tests should be run against the port production environment(using a testing org). </h4>

### Core testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync finishes successfully
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Scheduled resync able to abort existing resync and start a new one
- [ ] Tested with at least 2 integrations from scratch
- [ ] Tested with Kafka and Polling event listeners
- [ ] Tested deletion of entities that don't pass the selector

### Integration testing checklist

- [ ] Integration able to create all default resources from scratch
- [ ] Resync able to create entities
- [ ] Resync able to update entities
- [ ] Resync able to detect and delete entities
- [ ] Resync finishes successfully
- [ ] If new resource kind is added or updated in the integration, add example raw data, mapping and expected result to the `examples` folder in the integration directory.
- [ ] If resource kind is updated, run the integration with the example data and check if the expected result is achieved
- [ ] If new resource kind is added or updated, validate that live-events for that resource are working as expected
- [ ] Docs PR link [here](#)

### Preflight checklist

- [ ] Handled rate limiting
- [ ] Handled pagination
- [ ] Implemented the code in async
- [ ] Support Multi account

## Screenshots

Include screenshots from your environment showing how the resources of the integration will look.

## API Documentation

Provide links to the API documentation used for this integration.

### CODE REVIEW

File: port_ocean/helpers/retry.py [hunk 1]
```diff
@@ -55,14 +55,14 @@ class RetryTransport(httpx.AsyncBaseTransport, httpx.BaseTransport):
             HTTPStatus.GATEWAY_TIMEOUT,
         ]
     )
-    MAX_BACKOFF_WAIT = 60
+    MAX_BACKOFF_WAIT = 60 * 5  # 5 minutes
```
Comment (Tankilevitch) [inline port_ocean/helpers/retry.py:58]:
```suggestion
    MAX_BACKOFF_WAIT_IN_SECONDS = 60 * 5
```

Comment (Tankilevitch) [inline port_ocean/helpers/retry.py:58]: changing this here will effect both port client and all integrations clients, is it intended?

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
missing changelog and version bump

**Review by Tankilevitch — APPROVED** [review]

File: port_ocean/clients/port/utils.py [hunk 2]
```diff
@@ -34,7 +34,11 @@ def _get_http_client_context(port_client: "PortClient") -> httpx.AsyncClient:
     if client is None:
         client = OceanAsyncClient(
             TokenRetryTransport,
-            transport_kwargs={"port_client": port_client},
+            transport_kwargs={
+                "port_client": port_client,
+                "max_backoff_wait": 60 * 5,
```
Comment (erikzaadi) [inline port_ocean/clients/port/utils.py:39]: Extract the `5` to a named const

File: port_ocean/clients/port/utils.py [hunk 3]
```diff
@@ -16,7 +16,7 @@
 # period of time, before raising an exception.
 # The max_connections value can't be too high, as it will cause the application to run out of memory.
 # The max_keepalive_connections can't be too high, as it will cause the application to run out of available connections.
-PORT_HTTP_MAX_CONNECTIONS_LIMIT = 200
+PORT_HTTP_MAX_CONNECTIONS_LIMIT = 100
```
Comment (erikzaadi) [inline port_ocean/clients/port/utils.py:19]: Wonder if it's worth having this configurable by env var for quick patches/tests later on

**Review by erikzaadi — APPROVED** [review]
1 NITs

**Pull request merged**